server/plugin.go

```diff
@@ -338,6 +358,59 @@ func (p *Plugin) handleAction(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (p *Plugin) updateKVStore(topicName string) error {
+	var topics = SNSTopics{}
+	val, err := p.API.KVGet(p.ChannelID)
+	if err != nil {
+		p.API.LogError("Unable to Get from KV Store")
+		return err
+	}
+	if val == nil {
+		// Initialise the map before first assignment
+		topics.Topics = make(map[string]bool)
+		topics.Topics[topicName] = true
+	} else {
+		unmarshalErr := json.Unmarshal(val, &topics)
+		if unmarshalErr != nil {
+			p.API.LogError("Unmarshal failed for existing Topics in KV Store")
+			return unmarshalErr
+		}
+		topics.Topics[topicName] = true
+	}
+	b, marshalErr := json.Marshal(topics)
+	if marshalErr != nil {
+		p.API.LogError("Unable to marshal Topics struct to JSON")
+		return marshalErr
+	}
+	p.API.KVSet(p.ChannelID, b)
```

Comment:
question for @cpoile - when setting the key, should we be prepending anything to the channel ID?  perhaps `topics_`?

Reply:
It depends -- you usually use a prefix if there are multiple "tables" and you want the ability to delete all the keys in that "table". Since we don't want to break this in the future by adding a prefix, let's do that. 

@msvbhat can you add a prefix to all keys, something like `const topicListPrefix = "topicsInChan_"`, and the KVSet above would be something like `p.API.KVSet(topicListPrefix + p.ChannelID)`
Then the rest of the calls to KVSet & KVGet would use the same prefix.

Reply:
Thanks for the additional feedback @cpoile.